src/utils.ts

```diff
@@ -360,3 +351,27 @@ export const TOOLBOX = {
     }
   ]
 };
+
+// Defining a Blockly Theme in accordance with the current JupyterLab Theme. 
+export function define_jupyter_theme() : Blockly.Theme{
+  Blockly.registry.unregister('theme', 'jupyterlab');
+
+  var jupyterlab_theme = Blockly.Theme.defineTheme('jupyterlab', {
+  'base': Blockly.Themes.Classic,
+  'componentStyles': {
+    'workspaceBackgroundColour': 'var(--jp-layout-color0)',
+    'toolboxBackgroundColour': 'var(--jp-layout-color2)',
+    'toolboxForegroundColour':  'var(--jp-ui-font-color0)',
+    'flyoutBackgroundColour': 'var(--jp-border-color2)',
+    'flyoutForegroundColour': 'var(--jp-layout-color3)',
+    'flyoutOpacity': 1,
+    'scrollbarColour': 'var(--jp-border-color0)',
+    'insertionMarkerColour': 'var(--jp-warn-color-active)',
+    'insertionMarkerOpacity': 0.3,
+    'scrollbarOpacity': 0.4,
+    'cursorColour': 'var(--jp-scrollbar-background-color)',
+  }
+});
+
+  return jupyterlab_theme;
+}
```

Comment:
Now, that we added the JupyterLab variables, it is not necessary to register a new theme everytime we need to compute the CSS variables. Instead, we can register the theme only once when loading the extension by doing something like this:

```suggestion
// Defining a Blockly Theme in accordance with the current JupyterLab Theme.
const jupyterlab_theme = Blockly.Theme.defineTheme('jupyterlab', {
  'base': Blockly.Themes.Classic,
  'componentStyles': {
    'workspaceBackgroundColour': 'var(--jp-layout-color0)',
    'toolboxBackgroundColour': 'var(--jp-layout-color2)',
    'toolboxForegroundColour':  'var(--jp-ui-font-color0)',
    'flyoutBackgroundColour': 'var(--jp-border-color2)',
    'flyoutForegroundColour': 'var(--jp-layout-color3)',
    'flyoutOpacity': 1,
    'scrollbarColour': 'var(--jp-border-color0)',
    'insertionMarkerColour': 'var(--jp-warn-color-active)',
    'insertionMarkerOpacity': 0.3,
    'scrollbarOpacity': 0.4,
    'cursorColour': 'var(--jp-scrollbar-background-color)',
  }
});

export const THEME: Blockly.Theme = jupyterlab_theme;

```